lib/system/alloc.nim

```diff
@@ -65,16 +64,20 @@ type
     freeList: ptr FreeCell
     free: int            # how many bytes remain
     acc: int             # accumulator for small object allocation
-    when defined(cpu32):
-      align: int
-    data: AlignType      # start of usable memory
+    when defined(nimAlignPragma):
+      data {.align: MemAlign.}: UncheckedArray[byte]      # start of usable memory
+    else:
+      data: UncheckedArray[byte]
```

Comment:
Dangerous for bootstrapping purposes. Why not keep the old, more correct variant for it?

Reply:
Yea, I thought that you might comment on this part. Well, I changed the value for `MemAlign` from 8 to 16, this is also alignment guarantee for malloc in C. This will also work for many SIMD data types if native optimizations are not turned on.

Why not keep the old, more correct variant? Well the old variant is with the new value of `MemAlign` not correct anymore. It is true that alignment constraints aren't met during bootstrapping. But the question is: "Are the alignment constraints necessary during compilation?", and I would say: "No". There are no SIMD types used in the compiler yet. And if there are ever SIMD types used in the compiler, the branch without the align pragme should be dead by then.
I also wouldn't call it "dangerous", everything that might happen is, during Bootstrapping on 32 bit systems, float64 operations might be not as fast as they could be.